File: 283. Move Zeroes.cpp
283. Move Zeroes 
Difficulty: Easy
Given an array nums, write a function to move all 0’s to the end of it while maintaining the relative order of the non-zero elements.

For example, given nums = [0, 1, 0, 3, 12], after calling your function, nums should be [1, 3, 12, 0, 0].

Note:
You must do this in-place without making a copy of the array.
Minimize the total number of operations.

class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int len = nums.size();
        for(int i = 0; i < len - 1; i++) {
            if(nums[i] == 0) {
                for(int j = i + 1; j < len; j++) {
                    if(nums[j] != 0) {
                        swap(nums[i], nums[j]);
                        break;
                    }
                }
            }
        }
    }
};

File: Swift/283. Move Zeroes.swift
// 283. Move Zeroes
// 16 ms
var j = 0
for i in 0..<nums.count {
    if nums[i] != 0 {
        nums[j] = nums[i]
        j += 1
    }
}
for i in j..<nums.count {
    nums[i] = 0
}